addons/account/models/account_bank_statement.py

```diff
@@ -597,13 +600,17 @@ def _get_common_sql_query(self, overlook_partner = False, excluded_ids = None, s
         acc_type = "acc.internal_type IN ('payable', 'receivable')" if (self.partner_id or overlook_partner) else "acc.reconcile = true"
         select_clause = "SELECT aml.id "
         from_clause = "FROM account_move_line aml JOIN account_account acc ON acc.id = aml.account_id "
-        where_clause = """WHERE aml.company_id = %(company_id)s  
-                                AND (
-                                        (aml.statement_id IS NULL AND aml.account_id IN %(account_payable_receivable)s 
-                                        AND aml.payment_id IS NOT NULL) 
-                                    OR 
-                                        ("""+acc_type+""" AND aml.reconciled = false)
-                                    )"""
+        account_clause = ''
+        if self.journal_id.default_credit_account_id and self.journal_id.default_debit_account_id:
+            account_clause = "aml.account_id IN %(account_payable_receivable)s AND"
```

Comment:
account_clause = "(aml.statement_id IS NULL AND aml.account_id IN %(account_payable_receivable)s AND aml.payment_id IS NOT NULL) OR"

where_clause = """WHERE aml.company_id = %(company_id)s
                           AND (
                                     """ + account_clause + """
                                     ("""+acc_type+""" AND aml.reconciled = false)
                           )"""